### DISCUSSION TITLE
List of commands for device parameters like noise and dynamic window?

### DISCUSSION THREAD

**Discussion by spw223** [IBM/aihwkit#480] [Q&A]
I have been looking for a comprehensive explanation of how to change noise and dynamic window, etc. of the device parameters, but much of the documentation is sparsely notated.

I found some settings to adjust noise in the examples, but I cannot find any mention of dynamic window parameters, nor can I find a comprehensive list of commands (with detailed explanations ideally) for tweaking device parameters. Is there such a list, or similar resource?

Thank you

**Comment by maljoras** [comment]
Hi @spw223:
Many thanks for your questions. There are many setting of the device parameters which are all described in the API documentation. They are defined in the `aihwkit.simulator.config` directory and you can search for the names in the API documentation to get detailed description of all settings.  The parameter depend e.g. on the device you are using for analog training.  

I don't know what you mean by "dynamic window" maybe you could specify somewhat. For instance, we have a input / output ratio that is fixed, which is related to the dynamic range of the mat-vec.  These parameter can be specified in the `rpu_config.forward` (IOParameter).
